### ISSUE TITLE
require() statements in npm distribution

### ISSUE THREAD

**Issue by jondavidjohn** [OnsenUI/OnsenUI#979]
I'm attempting to use onsenui via npm in a project.  I'm using browserify, and when I attempt to

    require('onsenui/js/onsenui');

I get the following error message

```
Error: Cannot find module './core.js' from '/Users/user/project/node_modules/onsenui/js'
    at /Users/user/project/node_modules/browserify/node_modules/resolve/lib/async.js:55:21
    at load (/Users/user/project/node_modules/browserify/node_modules/resolve/lib/async.js:69:43)
    at onex (/Users/user/project/node_modules/browserify/node_modules/resolve/lib/async.js:92:31)
    at /Users/user/project/node_modules/browserify/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:82:15)
```

This is happening because there are 2-3 `require(...)` calls in the distributed npm package, which is probably a mistake?

This is happening on the `2.0.0-alpha.6` release.

**Comment by jondavidjohn** [comment]
For some context, I'm attempting to follow [this blog post](http://onsen.io/blog/tutorial-create-instagram-gallery-app-with-onsenui-react/) about using onsenui with react, and additionally I found you do not package `ons-core` individually with your npm distribution which would also be an appreciated addition to the npm package.

**Comment by jondavidjohn** [comment]
I tracked it down to be in [core/vendor/promise-6.1.0.js](https://github.com/OnsenUI/OnsenUI/blob/f0c87264c5bcd38100e1d600ef414e66a3cb3907/core/vendor/promise-6.1.0.js#L172).

**Comment by argelius** [comment]
Thanks a lot @jondavidjohn !

I have fixed the promise polyfill as well as some other scripts where `module.exports` was changed.

I released a new version (`2.0.0-alpha.7`) and tried with the following script:

```javascript
var onsen = require('onsenui/js/onsenui');

onsen.ready(function() {
  onsen.notification.alert({message: 'Onsen UI works with browserify!'});
});
```

And browserified with:

```bash
$ browserify main.js -o bundle.js -d
```

which worked fine when I ran it in the browser.

There are still some issues with polluting the global scope that I will address later. It will require a pretty (very) large refactor but I will do it in the coming days.

By the way, we have renamed some files: `ons-core.js` has been renamed to `onsenui.js` while we have split the AngularJS code from `onsenui.js` into `angular-onsenui.js`.

Please report any other issues you might find! It's really appeciated :)

**Comment by argelius** [comment]
I have closed this, but please feel free to open it again.
